app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -233,4 +292,37 @@ def extract_ems_ref_id(href)
       href && href.split("/").last
     end
   end
+
+  class Cacher
```

Comment:
@borod108 I'm not sure why you're creating your own cache object only to store it in the application cache (`Rails.cache`). Any specific reasoning?

Perhaps `class Cacher < ActiveSupport::Cache::Store` and subclass any custom logic you need. There are a [variety of other build cache mechanisms to choose from](http://api.rubyonrails.org/v5.0.0.1/classes/ActiveSupport/Cache/Store.html), as well.